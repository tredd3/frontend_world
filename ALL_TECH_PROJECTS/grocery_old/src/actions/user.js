import {
  SET_USER,
  SET_NEW_AUTH_TOKEN
} from "./constants";
import API_ROUTES from '../helper/network/api-routes';
import {
  fetchApi
} from '../helper/network/fetch';
import store from '../store';
let NewaccessToken;

export const setUser = (data) => {
  return {
    type: SET_USER,
    data: data
  }
}

export const setNewToken = (data) => {
  return {
    type: SET_NEW_AUTH_TOKEN,
    data: data
  }
}

export const addUser = data => {
  return dispatch => {
    return fetchApi({
      url: API_ROUTES.addUser,
      body: data
    }).then(response => {
      dispatch(setUser({ userId: response.Data.userId }));
    }).catch(e => {

    })
    // dispatch(setUser({
    //   userId: "9953"
    // }));
  }
}



export const updateUser = data => {
  return (dispatch, getState) => {
    let nameArr = data.name.split(" ");
    let firstName = nameArr[0];
    const reducer = (accumulator, currentValue, i) => { return (i === 0) ? '' : accumulator + ' ' + currentValue };
    let lastName = nameArr.length !== 1
      ? nameArr.length !== 2
        ? nameArr.reduce(reducer, null)
        : nameArr[1]
      : ""
    fetchApi({
      url: API_ROUTES.updateUser,
      body: {
        userId: getState().user.userId,
        emailId: data.email,
        dob: data.dob,
        gender: data.gender,
        firstName,
        lastName,
        mobileNumber: data.phoneNumber
      }
    })
      .then(response => {
        let user = {
          ...getState().user,
          ...data
        }
        localStorage.setItem('user', JSON.stringify(user));
        return dispatch(setUser({
          ...user
        }));
      })
      .catch(e => {
        console.log("Error", e)
      })
  }
}

export const getUser = data => {
  return dispatch => {
    fetchApi({ url: API_ROUTES.getUserDetail, body: {} })
      .then(response => {
        let webApp = process.env.APP_TYPE == "web" ? true : false;
        let user = {
          customerName: response.Data.customerName,
          firstName: response.Data.firstName,
          lastName: response.Data.lastName,
          userId: response.Data.userId,
          phoneNumber: response.Data.mobileNumber,
          login: true,
          gender: response.Data.gender,
          dob: response.Data.dob,
          email: response.Data.emailId,
        }
        if (webApp) {
          user.access_token = data.access_token;
          user.refresh_token = data.refresh_token;
          user.uuid = data.uuid;
          user.expiresAt = data.expiresAt;
        } else {
          user = { ...store.getState().user, ...user };
        }
        localStorage.setItem('user', JSON.stringify(user));
        return dispatch(setUser({
          ...user
        }));
      })
      .catch(e => {
        let MessageCode = e.Control && e.Control.MessageCode
        if (MessageCode && MessageCode === 404) {
          if (data.history) {
            data.history.push("/account/address/add/addUder");
          }
          // fetchApi({
          //   url: API_ROUTES.addUser,
          //   body: { mobileNumber: "7021143366", firstName: "MyJio", lastName: "User" }
          // }).then(response => {
          //     dispatch(setUser({ userId: response.Data.userId }));
          //   })
          //   .catch(e => {

          //   }) 
        }
      })
  }
}

const addUserToBackend = data => {

}

export const getNewAccessToken = () => {
  return (dispatch, getState) => {
    const AUTH_KEY = process.env.AUTHORISATION_VALUE;
    var params = {
      grant_type: "refresh_token",
      refresh_token: getState().user.refresh_token
    }
    var formBody = [];
    for (var property in params) {
      var encodedKey = encodeURIComponent(property);
      var encodedValue = encodeURIComponent(params[property]);
      formBody.push(encodedKey + "=" + encodedValue);
    }
    formBody = formBody.join("&");
    fetch(API_ROUTES.getNewAccessToken, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
        "Authorization": `Basic ${AUTH_KEY}`,
      },
      body: formBody
    }).then(response => response.json())
      .then(response => {
        let userNewTokens = {
          access_token: response.access_token,
          refresh_token: response.refresh_token,
          expiresAt: Date.now() + (response.expires_in * 1000),
        }
        localStorage.setItem('user', JSON.stringify({ ...getState().user, ...userNewTokens }));
        return dispatch(setNewToken(userNewTokens));
      })
      .catch(e => {
        console.log("ERROR HAPPENED During intializing User ")
      })
  }
}
