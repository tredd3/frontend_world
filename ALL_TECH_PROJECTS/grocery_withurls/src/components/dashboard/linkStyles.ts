import { css } from '@emotion/core';

const baseStyles = css`
  text-decoration: none;
  outline: none;
`;

export default css`
  ${baseStyles};

  :hover, :focus, :visited, :active {
    ${baseStyles}
  }
`;
