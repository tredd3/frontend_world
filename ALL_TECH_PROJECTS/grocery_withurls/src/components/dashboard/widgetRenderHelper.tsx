import React from 'react';
import HorizontalScrollableIconsWidget from './widgets/HorizontalScrollableIcons';
import PagerWidget from './widgets/Pager';
import ThinCarouselBannerWidget from './widgets/ThinCarouselBanner';
import BigSquareCarouselBannerWidget from './widgets/BigSquareCarouselBanner';
import HorizontalProductsWidget from './widgets/HorizontalProducts';
import SmallSquareCarouselBannerWidget from './widgets/SmallSquareCarouselBanner';
import W4xWidget from './widgets/W4xWidget';
import W9xWidget from './widgets/W9xWidget';
import W4xWithBannerWidget from './widgets/W4xWithBanner';
import W9xWithBannerWidget from './widgets/W9xWithBanner';
import {
  HorizontalScrollableIcons,
  Pager,
  ThinCarouselBanner,
  BigSquareCarouselBanner,
  SmallSquareCarouselBanner,
  HorizontalProducts,
  W4x,
  W9x,
  W4xWithBanner,
  W9xWithBanner,
  Widget
} from '../../services/dashboard2';

export const renderWidget = (widgetData: Widget) => {
  const { type } = widgetData;

  switch (type) {
    case 1:
      return <HorizontalScrollableIconsWidget widgetData={widgetData as HorizontalScrollableIcons} />;

    case 3:
      return <PagerWidget widgetData={widgetData as Pager} />;

    case 4:
      return <ThinCarouselBannerWidget widgetData={widgetData as ThinCarouselBanner} />;

    case 5:
      return <BigSquareCarouselBannerWidget widgetData={widgetData as BigSquareCarouselBanner} />;

    case 6:
      return <SmallSquareCarouselBannerWidget widgetData={widgetData as SmallSquareCarouselBanner} />;

    case 7:
      return <HorizontalProductsWidget widgetData={widgetData as HorizontalProducts} />;

    case 11:
      return <W4xWidget widgetData={widgetData as W4x} />;

    case 12:
      return <W9xWidget widgetData={widgetData as W9x} />;

    case 15:
      return <W4xWithBannerWidget widgetData={widgetData as W4xWithBanner} />;

    case 16:
      return <W9xWithBannerWidget widgetData={widgetData as W9xWithBanner} />;

    default:
      return null;
  }
};

