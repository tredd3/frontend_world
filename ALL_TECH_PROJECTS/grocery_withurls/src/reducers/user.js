import {
  SET_USER, NEW_USER, SET_NEW_AUTH_TOKEN, ORDER_RATED
} from '../actions/constants';

let initialState = {
  // eslint-disable-next-line @typescript-eslint/camelcase
  access_token: null,
  // eslint-disable-next-line @typescript-eslint/camelcase
  refresh_token: null,
  firstName: '',
  lastName: '',
  userId: 0,
  phoneNumber: '',
  login: false,
  xApiKey: null,
  newUser: false,
  newUserErrorMessage: '',
  orderRated: false
};
const placeOrderDataObj = sessionStorage.getItem('placeOrderDataObj');
let sessionUser = null;
if (placeOrderDataObj) {
  sessionUser = JSON.parse(placeOrderDataObj).user;
}
initialState = sessionUser || initialState;

export default function user(state = initialState, action) {
  switch (action.type) {
    case SET_USER:
      return { ...state, ...action.data };
    case SET_NEW_AUTH_TOKEN:
      return { ...state, ...action.data };
    case NEW_USER:
      return { ...state, ...action.data };
    case ORDER_RATED:
      return { ...state, ...action.data };
    default:
      return state;
  }
}
