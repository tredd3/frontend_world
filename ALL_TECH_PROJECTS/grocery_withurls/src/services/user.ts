import promiseMemoize from 'promise-memoize';
import makeApiCall from './helpers/make-api-call';
import { apiRoutes } from './api-routes';
import { formatDate } from '../helpers/functional';
import { pivotAPIUser } from './helpers/pivots';
import { APIUser, UserPivot } from '../types';
import { genderStringToCode } from './helpers/generic';

export const getUser = promiseMemoize(async (): Promise<UserPivot> => {
  const user = await makeApiCall<{}, APIUser>(apiRoutes.getUserDetail);
  return pivotAPIUser(user);
});

type UpdateUserArgs = {
  userId?: number;
  firstName?: string;
  lastName?: string;
  email?: string;
  gender?: UserPivot['gender'];
  dob?: Date;
  lat?: number;
  lng?: number;
  pincode?: number;
  city?: string;
  storeId?: number;
  storeName?: string;
};

export const updateUser = async ({
  firstName, lastName, email, gender, dob, lat, lng, pincode, city, storeId, storeName
}: UpdateUserArgs) => {
  const userFromApi = await getUser();
  const { userId } = userFromApi;
  const isValidStoreId = storeId !== undefined;
  const isValidStoreName = storeName !== undefined;

  await makeApiCall(apiRoutes.updateUser, {
    body: {
      userId,
      ...(firstName && { firstName }),
      ...(lastName && { lastName }),
      ...(email ? { emailId: email } : { emailId: '' }),
      ...(gender && { gender: genderStringToCode(gender) }),
      ...(dob ? { dob: formatDate(dob) } : { dob: '' }),
      preferences: {
        ...userFromApi.preferences,
        ...(pincode && { pincode }),
        ...(lat && { latitude: lat }),
        ...(lng && { longitude: lng }),
        ...(isValidStoreId && { storeId }),
        ...(city && { city }),
        ...(isValidStoreName && { storeName })
      }
    }
  });

  getUser.clear();
};

type AddUserArgs = {
  firstName: string;
  lastName: string;
};

export const addUser = ({ firstName, lastName }: AddUserArgs) => (
  makeApiCall<AddUserArgs, void>(apiRoutes.addUser, {
    body: {
      firstName,
      lastName
    }
  })
);
