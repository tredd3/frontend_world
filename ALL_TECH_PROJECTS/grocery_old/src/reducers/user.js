import { SET_USER, SET_NEW_AUTH_TOKEN } from '../actions/constants';

let initialState = {
  access_token: null,
  refresh_token: null,
  firstName: "",
  lastName: "",
  userId: 0,
  phoneNumber: '',
  login: false,
  xApiKey: null
}
initialState = JSON.parse(localStorage.getItem('user')) || initialState;

export default function user(state = initialState, action) {
  switch (action.type) {
    case SET_USER:
      return { ...state, ...action.data }
    case SET_NEW_AUTH_TOKEN:
      return { ...state, ...action.data }
    default:
      return state;
  }
}
