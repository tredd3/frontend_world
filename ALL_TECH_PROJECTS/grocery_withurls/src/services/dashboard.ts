import promiseMemoize from 'promise-memoize';
import makeApiCall from './helpers/make-api-call';
import { apiRoutes } from './api-routes';
import { getUser } from './user';
import { APIDashboardProduct, DashboardProductPivot } from '../types';
import { pivotDashboardProduct } from './helpers/pivots';

type APIRequest = {
  bannerId: number;
  isFC: 1;
  latitude: number;
  longitude: number;
  mobileNo: number;
  size: number;
  type: number;
  subType: number;
  pageNum: number;
  pincode: number;
  storeId: number;
  userId: number;
};

export type APIResponse = {
  products: APIDashboardProduct[];
  Count: number;
};

type GetAllDashboardDataArgs = {
  bannerId: number;
  type: number;
  subType: number;
  pageNum: number;
}

export type GetAllDashboardDataReturn = {
  products: DashboardProductPivot[];
  totalCount: number;
}

export const getAllDashboardData = promiseMemoize(
  async ({
    bannerId,
    type,
    subType,
    pageNum
  }: GetAllDashboardDataArgs): Promise<GetAllDashboardDataReturn> => {
    const {
      userId,
      phoneNumber,
      preferences: {
        storeId, pincode, lat, lng
      }
    } = await getUser();
    const { products, Count } = await makeApiCall<APIRequest, APIResponse>(apiRoutes.viewAllDashboard, {
      body: {
        bannerId,
        isFC: 1,
        latitude: lat || 0,
        longitude: lng || 0,
        mobileNo: phoneNumber,
        pageNum,
        pincode: pincode || 0,
        size: 20,
        storeId: storeId || 0,
        subType,
        type,
        userId
      }
    });

    return { products: products.map(pivotDashboardProduct), totalCount: Count };
  },
  { resolve: 'json' }
);
