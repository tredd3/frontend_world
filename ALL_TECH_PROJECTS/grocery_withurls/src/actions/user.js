import { SET_USER, NEW_USER, SET_NEW_AUTH_TOKEN } from './constants';
import { checkServiceability } from './address';
import { getUser as getUserService, addUser as addUserService } from '../services/user';
import { getConfig } from '../services/config';
import updateUserAction from './update-user';
import { APIError } from '../services/types';

export const updateUser = updateUserAction;

export const setUser = data => ({
  type: SET_USER,
  data
});

export const newUser = data => ({
  type: NEW_USER,
  data
});

export const setNewToken = data => ({
  type: SET_NEW_AUTH_TOKEN,
  data
});

export const setNewUser = () => newUser({ newUser: true });

export const addUser = ({ firstName, lastName }) => async dispatch => {
  try {
    await addUserService({ firstName, lastName });
    getUserService.clear();
    const user = await getUserService();
    dispatch(setUser(user));
    dispatch(newUser({ newUser: false, newUserErrorMessage: '' }));
    try {
      await getConfig();
    } catch (e) {
      console.log(e.message);
    }
  } catch (e) {
    if (e instanceof APIError) {
      dispatch(newUser({ newUserErrorMessage: e.message }));
    }
  }
};

export const getUser = () => async dispatch => {
  try {
    const user = await getUserService();
    return [
      dispatch(setUser(user)),
      dispatch(checkServiceability({ pincode: user.preferences.pincode }))
    ];
  } catch (e) {
    return [
      dispatch(setUser({}))
    ];
  }
};
