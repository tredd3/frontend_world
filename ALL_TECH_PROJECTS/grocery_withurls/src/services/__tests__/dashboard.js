import mockDB from '../../mocks/db.json';
import { getUser } from '../user';
import { getAllDashboardData } from '../dashboard';

jest.mock('../../intents', () => ({
  getTokens: () => ({
    apiKey: 'dummy-api-key',
    accessToken: 'dummy-access-token'
  })
}));

describe('Get All Dashboard Data', () => {
  beforeEach(fetch.resetMocks);
  afterEach(() => {
    getUser.clear();
    getAllDashboardData.clear();
  });

  it('should return a properly formatted product array', async () => {
    fetch.mockResponseOnce(JSON.stringify(mockDB.user));
    fetch.mockResponseOnce(JSON.stringify(mockDB.getAllDashboard));

    const products = await getAllDashboardData({
      bannerId: 123, type: 1, subType: 2, pageNum: 1
    });
    expect(products).toMatchSnapshot();
  });
});
