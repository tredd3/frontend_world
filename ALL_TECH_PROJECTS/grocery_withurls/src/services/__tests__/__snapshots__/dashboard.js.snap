// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get All Dashboard Data should return a properly formatted product array 1`] = `
Object {
  "products": Array [
    Object {
      "brandName": "Pampers",
      "category": Object {
        "id": 10000925,
        "name": "Baby Diapers & Wipes",
      },
      "coupon": null,
      "id": 100035656,
      "images": Array [
        "https://api.reliancesmart.in//images/?name=images/h8f/hac/8994619981854.jpg",
      ],
      "isFc": 1,
      "mrp": 349,
      "name": "Pampers Active New Baby Diapers 24 count (Up to 5 kg)",
      "skuId": 10042760,
      "sp": 324.57,
    },
    Object {
      "brandName": "Pampers",
      "category": Object {
        "id": 10000925,
        "name": "Baby Diapers & Wipes",
      },
      "coupon": null,
      "id": 100092959,
      "images": Array [
        "https://api.reliancesmart.in//images/?name=images/h33/hb6/9049450446878.jpg",
      ],
      "isFc": 1,
      "mrp": 699,
      "name": "Pampers Premium Care Pants (XL) 24 count (12 - 17 kg)",
      "skuId": 10105624,
      "sp": 449,
    },
    Object {
      "brandName": "Pampers",
      "category": Object {
        "id": 10000925,
        "name": "Baby Diapers & Wipes",
      },
      "coupon": null,
      "id": 100009797,
      "images": Array [
        "https://api.reliancesmart.in//images/?name=images/h37/h36/9012645330974.jpg",
      ],
      "isFc": 1,
      "mrp": 699,
      "name": "Pampers Active Baby Diapers (S) 46 count (3 - 8 kg)",
      "skuId": 10011234,
      "sp": 650.07,
    },
    Object {
      "brandName": "Pampers",
      "category": Object {
        "id": 10000925,
        "name": "Baby Diapers & Wipes",
      },
      "coupon": null,
      "id": 100037052,
      "images": Array [
        "https://api.reliancesmart.in//images/?name=images/hbc/h22/9036312018974.jpg",
      ],
      "isFc": 1,
      "mrp": 699,
      "name": "Pampers Premium Care Pants (M) 38 count (7 - 12 kg)",
      "skuId": 10044428,
      "sp": 449,
    },
    Object {
      "brandName": "Pampers",
      "category": Object {
        "id": 10000925,
        "name": "Baby Diapers & Wipes",
      },
      "coupon": null,
      "id": 100037029,
      "images": Array [
        "https://api.reliancesmart.in//images/?name=images/h69/he4/9026458943518.jpg",
      ],
      "isFc": 1,
      "mrp": 1049,
      "name": "Pampers Baby Dry Pants (L) 64 count (9 - 14 kg)",
      "skuId": 10044404,
      "sp": 809,
    },
    Object {
      "brandName": "Vicks",
      "category": Object {
        "id": 10001209,
        "name": "Over The Counter Remedies",
      },
      "coupon": null,
      "id": 100039704,
      "images": Array [
        "https://api.reliancesmart.in//images/?name=images/h14/h5b/8995540697118.jpg",
      ],
      "isFc": 1,
      "mrp": 149,
      "name": "Vicks Baby Rub 50 ml",
      "skuId": 10047322,
      "sp": 141.55,
    },
    Object {
      "brandName": "Pampers",
      "category": Object {
        "id": 10000925,
        "name": "Baby Diapers & Wipes",
      },
      "coupon": null,
      "id": 100042967,
      "images": Array [
        "https://api.reliancesmart.in//images/?name=images/h7b/hdc/9026334359582.jpg",
      ],
      "isFc": 1,
      "mrp": 699,
      "name": "Pampers Baby Dry Pants (L) 46 count (9 - 14 kg)",
      "skuId": 10050442,
      "sp": 699,
    },
    Object {
      "brandName": "Pampers",
      "category": Object {
        "id": 10000925,
        "name": "Baby Diapers & Wipes",
      },
      "coupon": null,
      "id": 100043308,
      "images": Array [
        "https://api.reliancesmart.in//images/?name=images/h33/h09/9026332983326.jpg",
      ],
      "isFc": 1,
      "mrp": 699,
      "name": "Pampers Baby Dry Pants (M) 54 count (7 - 12 kg)",
      "skuId": 10050789,
      "sp": 499,
    },
  ],
  "totalCount": 8,
}
`;
